pkg/adapter/config/encryptionAdapter.go

```diff
@@ -2,46 +2,155 @@ package config
 	name common.SimpleDoguName,
-	value common.EncryptedDoguConfigValue,
+	encryptedValue common.EncryptedDoguConfigValue,
 ) (common.SensitiveDoguConfigValue, error) {
-	//TODO implement me
-	panic("implement me")
+	privateKeySecret, err := p.secrets.Get(ctx, string(name)+"-private", metav1.GetOptions{})
+	if err != nil {
+		return "", domainservice.NewNotFoundError(err, "could not get private key")
+	}
+	privateKey := privateKeySecret.Data["private.pem"]
+	keyPair, err := getKeyPairFromPrivateKey(privateKey, p.registry)
+	if err != nil {
+		return "", domainservice.NewInternalError(err, "could not get key pair for dogu %v", string(name))
+	}
+	decryptedValue, err := keyPair.Private().Decrypt(string(encryptedValue))
+	if err != nil {
+		return "", domainservice.NewInternalError(err, "could not decrypt encrypted value")
+	}
+	return common.SensitiveDoguConfigValue(decryptedValue), nil
 }
 
-func (p PublicKeyConfigEncryptionAdapter) DecryptAll(
-	ctx context.Context,
-	m map[common.SensitiveDoguConfigKey]common.EncryptedDoguConfigValue,
-) (map[common.SensitiveDoguConfigKey]common.SensitiveDoguConfigValue, error) {
-	//TODO implement me
-	panic("implement me")
+func getKeyPairFromPrivateKey(privateKey []byte, registry etcdRegistry) (*keys.KeyPair, error) {
+	keyProvider, err := resource.GetKeyProvider(registry)
+	if err != nil {
+		return &keys.KeyPair{}, domainservice.NewNotFoundError(err, "could not get key provider")
+	}
+	keyPair, err := keyProvider.FromPrivateKey(privateKey)
+	if err != nil {
+		return &keys.KeyPair{}, domainservice.NewInternalError(err, "could not get key pair")
+	}
+	return keyPair, nil
+}
+
+func (p PublicKeyConfigEncryptionAdapter) DecryptAll(ctx context.Context, entries map[common.SensitiveDoguConfigKey]common.EncryptedDoguConfigValue) (map[common.SensitiveDoguConfigKey]common.SensitiveDoguConfigValue, error) {
+	decryptedEntries := map[common.SensitiveDoguConfigKey]common.SensitiveDoguConfigValue{}
+	var decryptionErrors []error
+	keypairs, err := getKeyPairsFromMap(ctx, entries, p.secrets, p.registry)
+	if err != nil {
+		return decryptedEntries, domainservice.NewNotFoundError(err, "could not get key pair(s)")
+	}
+	for configKey, configValue := range entries {
+		doguname := string(configKey.DoguName)
+		keyPair := keypairs[doguname]
+		decryptedValue, err := keyPair.Private().Decrypt(string(configValue))
+		if err != nil {
+			decryptionErrors = append(decryptionErrors, domainservice.NewInternalError(err, "could not decrypt encrypted value for dogu %v", doguname))
+		}
+		decryptedEntries[configKey] = common.SensitiveDoguConfigValue(decryptedValue)
+	}
+	return decryptedEntries, errors.Join(decryptionErrors...)
+}
+
+func getKeyPairsFromMap(ctx context.Context, entries map[common.SensitiveDoguConfigKey]common.EncryptedDoguConfigValue, secrets secret, registry etcdRegistry) (map[string]*keys.KeyPair, error) {
+	var getKeyPairErrors []error
+	keypairs := map[string]*keys.KeyPair{}
+	for configKey, _ := range entries {
```

Comment:
🚫 **[checkstyle]** <gosimple> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>S1005: unnecessary assignment to the blank identifier